modules/post/linux/gather/openvpn_credentials.rb

```diff
@@ -57,7 +62,7 @@ def run
       return
     end
 
-    dump = cmd_exec('/bin/grep rw-p /proc/'"#{pid}"'/maps | sed -n \'s/^\([0-9a-f]*\)-\([0-9a-f]*\) .*$/\1 \2/p\' | while read start stop; do /usr/bin/gdb --batch-silent --silent --pid '"#{pid}"' -ex "dump memory '"#{tmp_path}#{pid}"'-$start-$stop.dump 0x$start 0x$stop"; done 2>/dev/null; echo $?')
+    dump = cmd_exec("/bin/grep rw-p /proc/#{pid}/maps | sed -n 's/^\([0-9a-f]*\)-\([0-9a-f]*\) .*$/\1 \2/p' | while read start stop; do /usr/bin/gdb --batch-silent --silent --pid #{pid} -ex \"dump memory #{tmp_path}#{pid}-$start-$stop.dump 0x$start 0x$stop\"; done 2>/dev/null; echo $?")
```

Comment:
Just a sanity check; Were the changes here intentional?

```
pid = 123
tmp_path = '/foo/bar'
```

Beore:

```diff
/bin/grep rw-p /proc/123/maps | sed -n 's/^\([0-9a-f]*\)-\([0-9a-f]*\) .*$/\1 \2/p' | while read start stop; do /usr/bin/gdb --batch-silent --silent --pid 123 -ex "dump memory /foo/bar123-$start-$stop.dump 0x$start 0x$stop"; done 2>/dev/null; echo $?
```

After
```
/bin/grep rw-p /proc/123/maps | sed -n 's/^([0-9a-f]*)-([0-9a-f]*) .*$/ /p' | while read start stop; do /usr/bin/gdb --batch-silent --silent --pid 123 -ex "dump memory /foo/bar123-$start-$stop.dump 0x$start 0x$stop"; done 2>/dev/null; echo $?
```

Diff

![image](https://github.com/user-attachments/assets/507b1414-afbf-4d8b-bf02-280a51ec68f1)

Reply:
Good catch. Fixed.